File: app/src/main/java/mchehab/com/androidlinearregression/MainActivity.java
package mchehab.com.androidlinearregression;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private EditText editTextNumber;
    private TextView textViewResult;
    private Button buttonPredict;

    private TFSimpleRegression tfSimpleRegression;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        editTextNumber = findViewById(R.id.editTextNumber);
        textViewResult = findViewById(R.id.textView);
        buttonPredict = findViewById(R.id.buttonPredict);

        tfSimpleRegression = new TFSimpleRegression(getApplicationContext());

        setButtonPredictListener();
    }

    private void setButtonPredictListener(){
        buttonPredict.setOnClickListener(e -> {
            float number = Float.parseFloat(editTextNumber.getText().toString());
            float result[] = tfSimpleRegression.predict(new float[]{number});
            textViewResult.setText(result[0] + "");
        });
    }
}

File: kotlin/src/main/java/mchehab/com/kotlin/MainActivity.kt
package mchehab.com.kotlin

import kotlinx.android.synthetic.main.activity_main.*
import android.support.v7.app.AppCompatActivity
import android.os.Bundle

class MainActivity : AppCompatActivity() {

    private var tfSimpleRegression: TFSimpleRegression? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        tfSimpleRegression = TFSimpleRegression(applicationContext)

        setButtonPredictListener()
    }

    private fun setButtonPredictListener() {
        buttonPredict!!.setOnClickListener { e ->
            val number = java.lang.Float.parseFloat(editTextNumber!!.text.toString())
            val result = tfSimpleRegression!!.predict(floatArrayOf(number))
            textView.text = result[0].toString() + ""
        }
    }
}
